### PR TITLE
Add PAE data and protein/PAE/LDDT plots

### PR DESCRIPTION
**Pull request by samuelmurail**
Hello,

First of all thanks for this wonderful soft !

- As in issue #89 , PAE data are missing, so I save the pae matrix as a numpy object (`.npy`).
- I also missed a lot the protein plots from `AlphaFold2_advanced.ipynb`, so I added it for each model with plDDT plot and PAE plot.

Maybe an option to save plots and PAE should be included in the `run()` function.

Here is an example of plot made with the `test_complex` :

3G5O_A_3G5O_B_all_plot_model_1_rank_1

Cheers

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Install poetry:
 ```shell
 curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
 # Make sure you have ~/.local/bin in PATH
-poetry config settings.virtualenvs.in-project true
+poetry config virtualenvs.in-project true
```
Comment (konstin) [inline Contributing.md:10]: Thanks!

File: colabfold/plot.py [hunk 2]
```diff
@@ -2,6 +2,55 @@
 
 import numpy as np
 from matplotlib import pyplot as plt
+from matplotlib import collections as mcoll
+import matplotlib
+from string import ascii_uppercase, ascii_lowercase
+
+pymol_color_list = [
```
Comment (konstin) [inline colabfold/plot.py:9]: Could you import that list like `from colabfold.colabfold import pymol_color_list`

File: colabfold/batch.py [hunk 3]
```diff
@@ -258,6 +265,21 @@ def predict_structure(
             )
             relaxed_pdb_path.write_text(relaxed_pdb_lines[r])
 
+        paes_path = result_dir.joinpath(
+            f"{prefix}_pae_{model_names[r]}_rank_{n + 1}.npy"
```
Comment (konstin) [inline colabfold/batch.py:269]: I think there's a json format from alphafold for that data that's still on my todo list add, but for this the numpy format is good

File: colabfold/plot.py [hunk 4]
```diff
@@ -61,3 +110,232 @@ def plot_lddt(
...
+    plddt=None,
+    pae=None,
+    Ls=None,
+    dpi=200,
+    best_view=True,
+    line_w=2.0,
+    show=False,
+):
+
+    use_ptm = False if pae is None else True
```
Comment (konstin) [inline colabfold/plot.py:255]:
```suggestion
    use_ptm = pae is not None
```

**Review by konstin — COMMENTED** [review]
I can't comment on the plots themselves, so I've only added some minor notes on the code

File: colabfold/plot.py (hunk 2 above)
Comment (samuelmurail) [reply colabfold/plot.py:9]: Well, in fact I noticed there was no import from `colabfold.py` directly in `batch.py` or `plot.py`, so I thought there was a reason for that, and I copied in `plot.py` few functions and variables like `kabsch`, `plot_pseudo_3D`, `add_text` and `plot_ticks` from `colabfold.py`.

So if `colabfold.py` is not deprecated, I should import all functions from colabfold.

Let me know what you think of it.

File: colabfold/batch.py (hunk 3 above)
Comment (samuelmurail) [reply colabfold/batch.py:269]: I think they are using the pickle library to export.

From deepmind alphafold:

```
`features.pkl` – A pickle file containing the input feature NumPy arrays used by the models to produce the structures.
```

I could easily implement it. However, there might be other useful model information to store in it.

File: colabfold/plot.py (hunk 2 above)
Comment (samuelmurail) [reply colabfold/plot.py]: In fact I realized that `colafold.py` was imported, so I remove all copied function and replace it by an import of `colabfold.colabfold`

File: colabfold/batch.py (hunk 3 above)
Comment (samuelmurail) [reply colabfold/batch.py]: I replaced `np.save()` with `pickle`

**Comment by martin-steinegger** [comment]
@samuelmurail thank you so much for contributing! Getting the raw data out is very important. Would it be okay for you to write the output file in a non-python specific format? I would prefer a text format. So that users that uses other languages like R can use the data easily as well.

**Comment by samuelmurail** [comment]
@martin-steinegger , I saved pae as `txt` files using `np.savetxt()` function.
As it may take a lot of disk space I've also add it in the zip file for each model.

**Comment by martin-steinegger** [comment]
@samuelmurail we added a json file containing the PAE information to the output. We implemented two formats the (1) AlphaFold Database JSON format for compatibility and (2) we a custom json with pae,ptmscore,plddt. Here is a Colab showing how to use the json file: https://github.com/sokrypton/ColabFold/blob/main/utils/plot_scores.ipynb
I am very sorry for not merging this PR. Thank you so much for contributing and bringing this to our attention!

**Pull request closed without merging**